sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -956,51 +960,43 @@ class RocksDB(
       acquiredThreadInfo = newAcquiredThreadInfo
       // Add a listener to always release the lock when the task (if active) completes
       Option(TaskContext.get()).foreach(_.addTaskCompletionListener[Unit] {
-        _ => this.releaseForThread(StoreTaskCompletionListener, newAcquiredThreadInfo.threadRef)
+        _ => this.release(StoreTaskCompletionListener, Some(newAcquiredThreadInfo))
       })
       logInfo(log"RocksDB instance was acquired by ${MDC(LogKeys.THREAD, acquiredThreadInfo)} " +
         log"for opType=${MDC(LogKeys.OP_TYPE, opType.toString)}")
     }
   }
 
   /**
-   * Function to release RocksDB instance lock for if the provided thread is the one
-   * that acquired it.
+   * Function to release RocksDB instance lock that allows for synchronized access to the state
+   * store instance. Optionally provide a thread to check against, and release only if provided
+   * thread is the one that acquired the lock.
    *
    * @param opType - operation type releasing the lock
-   * @param threadRef - thread reference to check against the acquired thread
+   * @param forThreadOpt - optional thread to check against acquired thread
    */
-  private def releaseForThread(opType: RocksDBOpType,
-    threadRef: WeakReference[Thread]): Unit = acquireLock.synchronized {
-    if (acquiredThreadInfo != null && acquiredThreadInfo.threadRef.get.isDefined) {
-      if (!threadRef.get.isDefined) {
-        logInfo(log"Thread reference is empty when attempting to release for" +
-          log" opType=${MDC(LogKeys.OP_TYPE, opType.toString)}, ignoring release." +
-          log" Lock is held by ${MDC(LogKeys.THREAD, acquiredThreadInfo)}")
-        return
-      }
+  private def release(opType: RocksDBOpType,
+    forThreadOpt: Option[AcquiredThreadInfo]): Unit = acquireLock.synchronized {
```

Comment:
we can just use a default arg here with value as `None`. that way u don't need to pass it from all the callers ?